File: service/recite-words/src/main/java/org/noodle/bean/user/info/UserInfoQueryRequest.java
package org.noodle.bean.user.info;


import org.noodle.beans.NoodleRequest;

/**
 * 用户查询服务
 * @author wanghy
 */
public class UserInfoQueryRequest extends NoodleRequest {
    /**
     * 用户昵称
     */
    private String nick;

    public String getNick() {
        return nick;
    }

    public void setNick(String nick) {
        this.nick = nick;
    }
}

File: client/ReciteWords/sdk/user/info/UserInfoQueryRequest.cs
using sdk;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ReciteWords
{
    public class UserInfoQueryRequest : NoodleRequest<UserInfoQueryDto>
    {
        public string Nick { get; set; }
        public override string ApiMethodName { get => "user.info.query"; }
        public override string ClientRequestNo { get ; set; }
        public override string ApiVersion { get => "1.0"; }

        public override void Validate()
        {
            if (string.IsNullOrEmpty(Nick))
            {
                throw new ArgumentException("用户昵称不能为空");
            }
        }
    }
}
